test/functional/provider/ruby_spec.lua

```diff
@@ -109,3 +111,24 @@ describe('ruby provider', function()
     eq(2, eval('1+1'))  -- Still alive?
   end)
 end)
+
+describe('rubyeval()', function()
+  it('evaluates ruby objects', function()
+    eq({1, 2, {['key'] = 'val'}}, funcs.rubyeval('[1, 2, {key: "val"}]'))
+  end)
+
+  it('errors out when given non-string', function()
+    eq('Vim(call):E474: Invalid argument', exc_exec('call rubyeval(10)'))
+    eq('Vim(call):E474: Invalid argument', exc_exec('call rubyeval(v:_null_dict)'))
+    eq('Vim(call):E474: Invalid argument', exc_exec('call rubyeval(v:_null_list)'))
+    eq('Vim(call):E474: Invalid argument', exc_exec('call rubyeval(0.0)'))
+    eq('Vim(call):E474: Invalid argument', exc_exec('call rubyeval(function("tr"))'))
+    eq('Vim(call):E474: Invalid argument', exc_exec('call rubyeval(v:true)'))
+    eq('Vim(call):E474: Invalid argument', exc_exec('call rubyeval(v:false)'))
+    eq('Vim(call):E474: Invalid argument', exc_exec('call rubyeval(v:null)'))
+  end)
+
+  it('accepts NULL string', function()
+    neq(0, exc_exec('call rubyeval($XXX_NONEXISTENT_VAR_XXX)'))
```

Comment:
I copied this from the python specs. It's not passing, but I'm not really sure what the purpose of the test is. It seems like it's expected to return an error, but it's just sending `rubyeval("")`, which in ruby just returns `nil`, not an error.

Reply:
It's specific to the python provider. `perleval` does not throw an error and returns 0.